File: Shame/slv.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    ifstream infile("output.txt");
    if (!infile) {
        cerr << "Error opening file for reading!" << endl;
        return 1;
    }

    string encrypted;
    if (!getline(infile, encrypted)) { // Read the encrypted line and check for failure
        cerr << "Error reading from file!" << endl;
        infile.close();
        return 1;
    }

    char xorkey = 'X';
    for (size_t i = 0; i < encrypted.size(); ++i) {
        encrypted[i] = encrypted[i] ^ xorkey; // XOR again with the key to decrypt
    }

    cout << "Decrypted flag: " << encrypted << endl;

    infile.close();
    return 0;
}

File: Shame/slv.py
def decrypt_xor(file_path, xor_key):
    try:
        # Open the file and read the encrypted content
        with open(file_path, 'r') as infile:
            encrypted = infile.read().strip()  # Read and strip newline or extra spaces

        # Decrypt using XOR operation
        decrypted = ''.join(chr(ord(c) ^ ord(xor_key)) for c in encrypted)
        
        # Print the decrypted flag
        print("Decrypted flag:", decrypted)
    except FileNotFoundError:
        print(f"Error: File '{file_path}' not found!")
    except Exception as e:
        print(f"An error occurred: {e}")

# Example usage
decrypt_xor("output.txt", 'X')
